### PR TITLE
fixed bug, added padExtent to utils

### PR DESCRIPTION
**Pull request by CBarrett2**

### CODE REVIEW

**Comment by CBarrett2** [comment]
Resolves #89

**Comment by jleft** [comment]
Do we need to consider other scale types? What if we wanted to pad a linear scale? Maybe not for now...

Any thoughts on a function like this, with an, optional, padding param? (Could use `fc.util.extent` internally.)
```js
// sc.util.extent(data, fields, padding); // For example:
sc.util.extent(data, ['date'], 60 * 60 * 24);
```

**Comment by CBarrett2** [comment]
I like both of those ideas @jleft, maybe a new issue once this is pulled? Also perhaps an option to pad by percentage rather than an absolute value?

File: src/assets/js/main.js [hunk 1]
```diff
@@ -189,7 +190,7 @@
             var tx = zoom.translate()[0];
             var ty = zoom.translate()[1];
 
-            var xExtent = fc.util.extent(data, ['date']);
+            var xExtent = sc.util.findPaddedExtent(fc.util.extent(data, ['date']), 60 * 60 * 24);
```
Comment (jleft) [inline src/assets/js/main.js]: Will there be excess whitespace at the start and end of the chart when using a line/area/point series?

File: src/assets/js/main.js [hunk 2]
```diff
@@ -262,8 +263,9 @@
 
     // Create navigation chart
     var yExtent = fc.util.extent(sc.util.filterDataInDateRange(data, fc.util.extent(data, 'date')), ['low', 'high']);
+    var xExtent = sc.util.findPaddedExtent(fc.util.extent(data, 'date'), 60 * 60 * 24);
```
Comment (jleft) [inline src/assets/js/main.js]: Will this create a gap between the extremes of the series and the edge of the navigator?

File: src/index.html [hunk 3]
```diff
@@ -69,8 +69,11 @@
     </div>
     <script src="assets/js/sc.js"></script>
     <script src="assets/js/util/filterDataInDateRange.js"></script>
+    <script src="assets/js/util/findPaddedExtent.js"></script>
     <script src="assets/js/data/feed/coinbase/webSocket.js"></script>
     <script src="assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js"></script>
+    <script src="assets/js/webSocket.js"></script>
```
Comment (jleft) [inline src/index.html]: Remove.

File: src/index.html [hunk 4]
```diff
@@ -69,8 +69,11 @@
...
+    <script src="assets/js/ohlc.js"></script>
```
Comment (jleft) [inline src/index.html]: Remove.

File: src/assets/js/util/findPaddedExtent.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+(function(d3, fc) {
+    'use strict';
+    sc.util.findPaddedExtent = function(extent, period) {
```
Comment (jleft) [inline src/assets/js/util/findPaddedExtent.js]: I'm not sure if this is really _finding_ the padded extent, maybe it's _offsetting_ the extent? I quite liked _padExtent_ too... maybe I'm being too pedantic!

File: src/assets/js/main.js (hunk 1 above)
Comment (CBarrett2) [inline src/assets/js/main.js]: There's some excess whitespace on the nav bar, at either end, but none on the main chart

File: src/assets/js/main.js [hunk 6]
```diff
@@ -280,8 +292,18 @@
...
+                    var extent = fc.util.extent(data, ['date']);
+                    var earliest = brush.extent()[0][0];
+                    var latest = brush.extent()[1][0];
+                    /*if (brush.extent()[1][0].getTime() === extent[1].getTime()) {
+                        earliest = new Date(earliest.getTime() - bufferPeriod);
+                    }
+                    if (brush.extent()[0][0].getTime() === extent[0].getTime()) {
+                        latest = new Date(latest.getTime() + bufferPeriod);
+                    }*/
+
```
Comment (djmiley) [inline src/assets/js/main.js]: Removed bracketed code.

File: src/assets/js/main.js [hunk 7]
```diff
@@ -262,8 +273,9 @@
 
     // Create navigation chart
     var yExtent = fc.util.extent(sc.util.filterDataInDateRange(data, fc.util.extent(data, 'date')), ['low', 'high']);
+    var xExtent = fc.util.extent(data, 'date');
     var navTimeSeries = fc.chart.linearTimeSeries()
-        .xDomain(fc.util.extent(data, 'date'))
+        .xDomain(xExtent)
         .yDomain(yExtent)
```
Comment (djmiley) [inline src/assets/js/main.js]: What do we gain with this? Also maybe `xDomain` before `yDomain`

File: src/assets/js/main.js [hunk 8]
```diff
@@ -92,7 +100,8 @@
 
     // Set Reset button event
     function resetToLive() {
-        timeSeries.xDomain(standardDateDisplay);
+        var extent = sc.util.paddedExtent(standardDateDisplay, bufferPeriod);
+        timeSeries.xDomain(extent);
         render();
```
Comment (djmiley) [inline src/assets/js/main.js]: `extent` -> `paddedExtent`?

File: src/assets/js/main.js [hunk 9]
```diff
@@ -100,7 +109,6 @@
 
     // Create main chart and set how much data is initially viewed
     var timeSeries = fc.chart.linearTimeSeries()
-        .xDomain(standardDateDisplay)
         .xTicks(6);
```
Comment (djmiley) [inline src/assets/js/main.js]: Why remove this line?

Comment (CBarrett2) [inline src/assets/js/main.js]: It paves the way for removing standardDateDisplay as a global variable.

Comment (djmiley) [inline src/assets/js/main.js]: Ok. Yes, I see what's happening here.

**Pull request closed without merging**